klasses/bootcamp_admissions_client_test.py

```diff
@@ -0,0 +1,159 @@
+"""
+Tests for the bootcamp_admission_client module
+"""
+from unittest.mock import Mock
+
+import pytest
+from django.conf import settings
+from rest_framework import status
+
+from klasses.bootcamp_admissions_client import BootcampAdmissionClient
+
+# pylint: disable=missing-docstring,redefined-outer-name,unused-argument
+
+pytestmark = pytest.mark.django_db
+
+JSON_RESP_OBJ = {
+    "user": "foo@example.com",
+    "bootcamps": [
+        {
+            "bootcamp_id": 1,
+            "bootcamp_title": "Entrepreneurship",
+            "klasses": [
+                {
+                    "klass_id": 13,
+                    "klass_name": "Class 2 (Student)",
+                    "status": "no_show",
+                    "is_user_eligible_to_pay": False
+                },
+                {
+                    "klass_id": 12,
+                    "klass_name": "Class 1",
+                    "status": "coming",
+                    "is_user_eligible_to_pay": False
+                }
+            ]
+        },
+        {
+            "bootcamp_id": 6,
+            "bootcamp_title": "Master of Law",
+            "klasses": [
+                {
+                    "klass_id": 16,
+                    "klass_name": "Class 1",
+                    "status": "scholarship_not_awarded",
+                    "is_user_eligible_to_pay": True
+                }
+            ]
+        }
+    ]
+}
+
+
+def create_mocked_get_response(status_code, json_obj):
+    """
+    Helper function to create a mocked response for the requests.get function
+    """
+    mocked_response = Mock()
+    mocked_response.status_code = status_code
+    mocked_response.json.return_value = json_obj
+    return mocked_response
```

Comment:
Not sure I understand. Here are a few benefits of that change as I see them:

1. If you have a mocked request/response that you use multiple times (you use the same 200 request/response in here 3 times), you can define it once and declare it as a fixture for any test case that uses it.
1. If you need to set custom JSON response data/side effects, you can do that. The gist actually has several examples: [1](https://gist.github.com/gsidebo/5d305c8def9399983242814606162c8e#file-1_requests_fixture-patch-L46), [2](https://gist.github.com/gsidebo/5d305c8def9399983242814606162c8e#file-1_requests_fixture-patch-L54), [3](https://gist.github.com/gsidebo/5d305c8def9399983242814606162c8e#file-1_requests_fixture-patch-L127)
1. The patching of `requests` no longer happens in `setup_test_data`, which makes more 
sense since patching `requests` doesn't have anything to do with the test data
1. That fixture can be reused in other test cases in the future